Sources/UtiliKit/ActiveLabel/ActiveLabel.swift

```diff
@@ -0,0 +1,285 @@
+    }
+    
+    // MARK: - Loading View Functions
+    func showLoadingViews() {
+        if loadingViews.isEmpty {
+            configureLoadingViews()
+        }
+        loadingViews.forEach({ $0.isHidden = false })
+    }
+    
+    func hideLoadingViews() {
+        loadingViews.forEach({ $0.isHidden = true })
+    }
+    
+    func configureLoadingViews() {
+        for i in 0..<Int(estimatedNumberOfLines) {
+            let view = UIView()
+            view.translatesAutoresizingMaskIntoConstraints = false
+            view.backgroundColor = loadingViewColor
+            
+            let maskLayer = createLayerMask()
+            maskLayer.add(maskAnimation(), forKey: "maskAnimation")
+            view.layer.mask = maskLayer
+            loadingViews.append(view)
+            
+            addSubview(view)
+            NSLayoutConstraint.activate([view.leadingAnchor.constraint(equalTo: leadingAnchor),
+                                         view.heightAnchor.constraint(greaterThanOrEqualToConstant: loadingLineHeight)])
+            
+            if i == 0 {
+                view.topAnchor.constraint(equalTo: topAnchor).isActive = true
+            } else {
+                let aboveView = loadingViews[i - 1]
+                view.topAnchor.constraint(equalTo: aboveView.bottomAnchor, constant: loadingLineVerticalSpacing).isActive = true
+            }
+            
+            if i + 1 == estimatedNumberOfLines {
+                view.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+                if finalLineTrailingInset != 0 {
+                    view.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -finalLineTrailingInset).isActive = true
+                } else if finalLineLength != 0 {
+                    view.widthAnchor.constraint(equalToConstant: finalLineLength).isActive = true
+                } else {
+                    view.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+                }
+            } else {
+                view.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+            }
+        }
+        
+        loadingViews.forEach({ $0.isHidden = isHidden })
+    }
+    
+    // MARK: - CA Helpers
+    func createLayerMask() -> CALayer {
+        let maskLayer = CAGradientLayer()
+        maskLayer.anchorPoint = .zero
+        maskLayer.startPoint = CGPoint(x: 0, y: 1)
+        maskLayer.endPoint = CGPoint(x: 1, y: 1)
+        
+        let lightColor = UIColor(white: 1, alpha: 0.5)
+        let darkColor = UIColor(white: 1, alpha: 1)
+        
+        maskLayer.colors = [lightColor.cgColor, darkColor.cgColor, lightColor.cgColor]
+        if showLoadingViewsInStoryboard && isDisplayingInStoryboard {
+            maskLayer.locations = [NSNumber(value: 0.4), NSNumber(value: 0.5), NSNumber(value: 0.6)]
+        } else {
+            maskLayer.locations = [NSNumber(value: 0.0), NSNumber(value: 0.1), NSNumber(value: 0.2)]
```

Comment:
What's the reasoning behind setting `maskLayer.locations` differently for storyboard rendering vs non-storyboard?

Reply:
Changing the locations allows you to see the gradient in the storyboard since normal use the graduated part is clipped by the view. I wanted the developer to be able to see what the gradient would look like as they were designing it.

Reply:
👍 Let's just add a comment that explains that reasoning.